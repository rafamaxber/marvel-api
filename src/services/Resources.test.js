const { generateApiUrl } = require('./Resources');

describe('Resources', () => {
  const urlApiParameters = {
    BASE_URL: 'http://url_base.com.br/',
    TIMESTAMP: '1513382563229', // valid timestamp
    PATH: 'characters',
    PUBLIC_KEY: '999999999999',
    PRIVATE_KEY: '888888888888',
  };

  describe('generateApiUrl', () => {
    test('Should be a function', () => {
      expect(typeof generateApiUrl).toBe('function');
    });

    test('Should return a string', () => {
      expect(typeof generateApiUrl({ ...urlApiParameters })).toBe('string');
    });

    test('Should return an merged url when receive a URL API, marvel private key and marvel public key', () => {
      expect(generateApiUrl({ ...urlApiParameters }))
        .toBe('http://url_base.com.br/characters?ts=1513382563229&apikey=999999999999&hash=eca518762091e8796d5c69e20c1d475b');
    });

    test('Should return error if some value in parameter is undefined or null', () => {
      expect(() => {
        generateApiUrl();
      }).toThrow();
    });
  });
});
