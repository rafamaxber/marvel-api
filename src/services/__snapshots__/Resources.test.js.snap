// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resources generateApiUrl Should throw message equal to: Is necerrary send all parameters! 1`] = `"Is necerrary send all parameters!"`;

exports[`Resources routes Should match with snapshot 1`] = `
Object {
  "characters": "characters",
}
`;

exports[`Resources urlApiParameters Should match with snapshot 1`] = `
Object {
  "BASE_URL": undefined,
  "PATH": "",
  "PRIVATE_KEY": undefined,
  "PUBLIC_KEY": undefined,
  "TIMESTAMP": "",
}
`;
